fil_fungible_token/src/token/mod.rs

```diff
@@ -1162,25 +1374,158 @@ mod test {
+
+        // gets an allowance error
+        match err {
+            TokenError::TokenState(StateError::InsufficientAllowance {
+                owner,
+                operator,
+                allowance,
+                delta,
+            }) => {
+                assert_eq!(owner, *TREASURY);
+                assert_eq!(operator, *ALICE);
+                assert_eq!(allowance, TokenAmount::zero());
+                assert_eq!(delta, burn_amount);
+            }
+            e => panic!("unexpected error {:?}", e),
+        };
 
         // balances didn't change
         assert_eq!(token.total_supply(), TokenAmount::from(400_000));
         assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::from(400_000));
         assert_eq!(token.allowance(TREASURY, ALICE).unwrap(), TokenAmount::zero());
     }
 
+    #[test]
+    fn it_allows_delegated_burns_by_resolvable_pubkeys() {
+        let mut token = new_token();
+
+        let mint_amount = TokenAmount::from(1_000_000);
+        let approval_amount = TokenAmount::from(600_000);
+        let burn_amount = TokenAmount::from(600_000);
+
+        // create a resolvable pubkey
+        let secp_address = &secp_address();
+        let secp_id = token.msg.initialize_account(secp_address).unwrap();
+
+        // mint the total amount
+        token.mint(TOKEN_ACTOR, TREASURY, &mint_amount, &Default::default()).unwrap();
+        // approve the burner to spend the allowance
+        token.increase_allowance(TREASURY, secp_address, &approval_amount).unwrap();
+        // burn the approved amount
+        token.burn(secp_address, TREASURY, &burn_amount).unwrap();
+
+        // total supply decreased
+        assert_eq!(token.total_supply(), TokenAmount::from(400_000));
+        // treasury balance decreased
+        assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::from(400_000));
+        // burner approval decreased
+        assert_eq!(token.allowance(TREASURY, secp_address).unwrap(), TokenAmount::zero());
+
+        // cannot burn non-zero again
+        let err = token.burn(secp_address, TREASURY, &burn_amount).unwrap_err();
+        // gets an allowance error
+        match err {
+            TokenError::TokenState(StateError::InsufficientAllowance {
+                owner,
+                operator,
+                allowance,
+                delta,
+            }) => {
+                assert_eq!(owner, *TREASURY);
+                assert_eq!(operator, Address::new_id(secp_id));
+                assert_eq!(allowance, TokenAmount::zero());
+                assert_eq!(delta, burn_amount);
+            }
+            e => panic!("unexpected error {:?}", e),
+        };
+        // balances unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(400_000));
+        assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::from(400_000));
+        assert_eq!(token.allowance(TREASURY, secp_address).unwrap(), TokenAmount::zero());
+
+        // but can burn zero
```

Comment:
This should fail (for consistency with denying transfer of zero with zero allowance)

Reply:
as above, this is addressed in #62